puppet/lib/puppet/util/puppetdb/http.rb

```
@@ -55,7 +55,7 @@ def self.action(path_suffix, &http_callback)
             response = nil
           elsif response.is_a? Net::HTTPNotFound
             Puppet.warning("Error connecting to #{url.host} on #{url.port} at route #{route}, error message received was '#{response.message}'. #{SERVER_URL_FAIL_MSG if server_url_config}")
-            response = nil
+            response = :notfound
```

Here we mark the results so later on we can return the right error.